:es6: Show more buttons

philwills opened the pull request:
## What does this change?
✨ The module for doing the show more button on fronts

## Does this affect other platforms - Amp, Apps, etc?
No

## Tested in CODE?
Yes

<!-- AB test? https://git.io/v1V0x -->
<!-- AMP question? https://git.io/v9zIE -->
<!-- Does this PR meet the contributing guidelines? https://git.io/v1VEJ -->

SiAdcock commented on static/src/javascripts/projects/facia/modules/ui/container-show-more.js:
```diff
@@ -0,0 +1,256 @@
+type ButtonState =
+    | typeof STATE_DISPLAYED
+    | typeof STATE_HIDDEN
+    | typeof STATE_LOADING;
```
Would it be more explicit to define this type as a straightforward union, without the `typeof`s?

```js
type ButtonState =
    | STATE_DISPLAYED
    | STATE_HIDDEN
    | STATE_LOADING;
```

philwills replied:
That's what I did initially, but it didn't type check. Will go back and check my sanity on that.

SiAdcock commented on static/src/javascripts/projects/facia/modules/ui/container-show-more.js:
```diff
@@ -0,0 +1,256 @@
+    | typeof STATE_HIDDEN
+    | typeof STATE_LOADING;
+
+const readPrefs = containerId => {
```
It would be super-awesome to add type annotations for all function params / expected return values 🙇  🙏

philwills replied:
Yea, started work on this a while ago and hadn't noticed that I hadn't added return types, will get on it.

SiAdcock commented on static/src/javascripts/projects/facia/modules/ui/container-show-more.js:
```diff
@@ -0,0 +1,256 @@
+};
+
+const itemsByArticleId = ($el: bonzo) =>
+    groupBy(qwery(ITEM_SELECTOR, $el), el =>
```
We can probably replace this reference to `qwery` with `$` and then drop the `qwery` dependency from this module

philwills commented:
@SiAdcock I think I've addressed your points including finding out quite a bit more about Flow in the process. I'd definitely appreciate some feedback on whether what I've done is sane.

SiAdcock commented on static/src/javascripts/projects/facia/modules/ui/container-show-more.js:
```diff
@@ -0,0 +1,271 @@
+    hidden: 'hidden',
+});
+
+type DisplayState = $Keys<typeof DISPLAY_STATE>;
```
[Nice](https://flow.org/en/docs/types/utilities/#toc-keys)! 🎉
